src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Bindings/WatchableCloudBlobStream.cs

```diff
@@ -153,5 +187,40 @@ public ParameterLog GetStatus()
                 return null;
             }
         }
+
+        private void Log()
+        {
+            if (_logged)
+            {
+                return;
+            }
+
+            if (_blob == null || _logger == null)
+            {
+                return;
+            }
+
+            try
+            {
+                List<string> statisticsList = new List<string>
+                {
+                    $"BlobName: {_blob.Name}",
+                    $"ContainerName: {_blob.Container.Name}",
+                    $"ETag: {_blob.Properties.ETag}",
+                    $"BlobType: {_blob.BlobType}",
+                    $"BytesWritten: {_countWritten}",
+                    $"ElapsedTime: {_timeWrite.Elapsed}",
+                };
+
+                string statistics = string.Join(", ", statisticsList);
+                string logMessage = $"WriteStream ({statistics})";
+                _logger.LogInformation(logMessage);
```

Comment:
change to `LogDebug`. Default log level for system logs which we can see in Kusto is `Debug`

Reply:
Log `_blobAccessLog`

Reply:
If this method is just for `_logger.LogDebug({_blobAccessLog})` , you can add this line close method.

Reply:
Can you update this implementation to be less allocation heavy? We should be able to generate a single string here instead of creating a list of 6 strings and then joining them together.